README.md

```diff
@@ -44,139 +70,134 @@ source /usr/share/virtualenvwrapper/virtualenvwrapper.sh
+### Make a Virtual Environment and Activate it
 
 ```bash
 mkvirtualenv -p python3 open-duck-mini-runtime
 workon open-duck-mini-runtime
 ```
 
-Clone this repository on your rasp, cd into the repo, then :
+### Install the Repository
 
+Clone the repository and install it in editable mode:
 ```bash
 git clone https://github.com/apirrone/Open_Duck_Mini_Runtime
 cd Open_Duck_Mini_Runtime
-git checkout v2
 pip install -e .
 ```
 
-In Raspberry Pi 5, you need to perform the following operations
-
+**For Raspberry Pi 5:** You may need to replace the GPIO library.
 ```bash
 pip uninstall -y RPi.GPIO
 pip install lgpio
 ```
 
 
-## Test the IMU
+### Xbox One Controller Setup
 
-```bash
-python3 mini_bdx_runtime/mini_bdx_runtime/raw_imu.py
-```
+1.  Turn on your Xbox One controller and put it in pairing mode by long-pressing the sync button.
+2.  On your Raspberry Pi, run the following commands:
+    ```bash
+    bluetoothctl
+    scan on
+    ```
+3.  Wait for the controller to appear, then pair, trust, and connect to it:
+    ```bash
+    pair <controller_mac_address>
+    trust <controller_mac_address>
+    connect <controller_mac_address>
+    ```
+    The controller's LED should stop blinking.
 
-You can also run `python3 scripts/imu_server.py` on the robot and `python3 scripts/imu_client.py --ip <robot_ip>` on your computer to check that the frame is oriented correctly. 
+4.  Test the connection:
+    ```bash
+    python3 mini_bdx_runtime/xbox_controller.py
+    ```
 
-> To find the ip address of the robot, run `ifconfig` on the robot
+## Hardware Configuration
 
-## Test motors
+### Speaker Wiring and Configuration
+Follow this Adafruit tutorial for wiring the speaker: [Adafruit MAX98357 I2S Class-D Mono Amp](https://learn.adafruit.com/adafruit-max98357-i2s-class-d-mono-amp?view=all).
 
-This will allow you to verify all your motors are connected and configured.
+> **Note:** For now, do not activate `/dev/zero` when prompted in the tutorial.
 
+## Testing and Calibration
+
+### Test the IMU
+
+Run a basic test to ensure the IMU is working:
 ```bash
-python3 scripts/check_motors.py
+python3 mini_bdx_runtime/raw_imu.py
 ```
 
-## Make your duck_config.json
-
-Copy `example_config.json` in the home directory of your duck and rename it `duck_config.json`.
+To visualize the IMU data, run the server on the robot and the client on your computer:
+```bash
+# On the robot
+python3 dev/hardware/imu_server.py
 
-`cp example_config.json ~/duck_config.json`
+# On your computer
+python3 dev/hardware/imu_client.py --ip <robot_ip>
+```
+> Use `ifconfig` on the robot to find its IP address.
 
-In this file, you can configure some stuff, like registering if you installed the expression features, installed the imu upside down or and other stuff. You also write the joints offsets of your duck here
+### Test Motors
+Verify that all motors are connected and configured correctly:
+```bash
+python3 dev/hardware/configure_all_motors.py
```

Comment:
Warning, this will reconfigure all the motors. It should not be necessary, and it will do weird things if they have not been mounted correctly.